pkg/object/infos.go

```diff
@@ -59,7 +59,7 @@ func UnstructuredToInfo(obj *unstructured.Unstructured) (*resource.Info, error)
 		// kyaml adds both annotations for the time being, so we need to remove them both
 		// before apply.
 		delete(annos, kioutil.PathAnnotation)
-		delete(annos, kioutil.LegacyPathAnnotation)
+		delete(annos, kioutil.LegacyPathAnnotation) //nolint:staticcheck
```

Comment:
Why are these new nolints needed?

Reply:
Build started failing because of these after Go 1.16 -> 1.17 bump in https://github.com/kubernetes/test-infra/pull/24709. See https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cli-utils/497/cli-utils-presubmit-master/1482230404432793600#1:build-log.txt%3A276